Tests/powershell-yaml.Tests.ps1

```diff
@@ -0,0 +1,198 @@
+﻿# Copyright 2016 Cloudbase Solutions Srl
+#
+#    Licensed under the Apache License, Version 2.0 (the "License"); you may
+#    not use this file except in compliance with the License. You may obtain
+#    a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+#    under the License.
+#
+
+$moduleName = "powershell-yaml"
+$modulePath = Resolve-Path "..\powershell-yaml.psm1"
+Import-Module $modulePath -DisableNameChecking
+
+
+InModuleScope $moduleName {
+
+    # Confirm-Equality is a helper function which acts like a DeepEquals
+    # with special attention payed to the specific types the yaml decoder
+    # can handle.
+    function Confirm-Equality {
+        Param(
+            [Parameter(Mandatory=$true)]$expected,
+            [Parameter(Mandatory=$true)]$got
+        )
+
+        # check for easy way out; this should work for all simple value types:
+        if ($expected -eq $got) {
+            return $true
+        }
+
+        # else; handle hashes and arrays specially:
+        if ($expected -is [System.Array]) {
+            if ($got -isnot [System.Array] -or ($expected.Count -ne $got.Count)) {
+                return $false
+            }
+
+            # just iterate through the elements of the array comparing each one:
+            for ($i = 0; $i -lt $expected.Count; $i = $i + 1) {
+                if ( !(Confirm-Equality $expected.Get($i) $got.Get($i)) ) {
+                    return $false
+                }
+            }
+
+            return $true
+        }
+
+        if ($expected -is [Hashtable]) {
+            if ($got -isnot [Hashtable] -or ($expected.Count -ne $got.Count)) {
+                return $false
+            }
+
+            # iterate through all the keys:
+            $expected.Keys | % {
```

Comment:
From my manual tests and reasoning about the thing; I'm confident it works well enough for our purposes.